### ISSUE TITLE
[NumberBox / TextBox] The clear (x) button is not keyboard accesible

### ISSUE THREAD

**Issue by niels9001** [microsoft/microsoft-ui-xaml#5999]
### Describe the bug

When entering text in a TextBox or NumberBox, the clear (x) button appears. There's no way to get to that button by using the keyboard alone, which was called out as an a11y in the PowerToys repo:

https://github.com/microsoft/PowerToys/issues/13460

### Steps to reproduce the bug

1. Select a TextBox or NumberBox
2. Enter a value, the clear button appears
3. Press tab to navigate to the clear button
4. Focus goes to the next control.

### Expected behavior

Pressing tab should move the focus to that button and clicking on enter should trigger the clear button click event.

### Screenshots

https://user-images.githubusercontent.com/87321414/135026886-1623fdfc-4162-4413-ae50-d55f0f06c615.mp4 

### NuGet package version

Microsoft.UI.Xaml 2.7.0

### Windows app type

- [X] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

May 2021 Update (19043)

**Comment by eugenegff** [comment]
Using keyboard after tabbing into NumberBox whole text is selected, so it could be erased by single Delete or Backspace. On the other hand, making Clear button IsTabStop=true will make navigation via Tab key over several NumberBox-es in a form a nightmare!

Not showing Clear button if NumberBox was focused via Tab seems to be the ideal solution.

**Comment by niels9001** [comment]
@eugenegff Agree!

Not sure what the solution should be, hopefully the WinUI team can provide feedback.

**Comment by gabbybilka** [comment]
Closing as by design as the clear all button is designed as a mouse accelerator. A user with keyboard input should use a key combo such as Alt+A to select all and delete the content.

cc: @chigy
